woof-code/rootfs-skeleton/usr/sbin/quicksetup

```diff
@@ -1181,33 +1315,45 @@ FLAG_CHANGED=""
 
 if [ "$SET_KEYBOARD" ];then
  #two variables COMBO_KEYBOARD and CHECK_NUMLOCK
- XKB_LAYOUT="`echo -n "$COMBO_KEYBOARD" | cut -f 1 -d ' ' | cut -f 1 -d "$TABCHAR"`"
+ [ -z "$COMBO_KEYBOARD" -a -f /tmp/xkbl ] && read COMBO_KEYBOARD </tmp/xkbl
+ read XKB_LAYOUT c <<<$COMBO_KEYBOARD
  FONTMAP=""; CODEPAGE=""
- FLAG_CHANGED="${FLAG_CHANGED}$(gettext 'Keyboard Layout'):yes|" #111020 no need to restart X.
  if [ -f /etc/X11/xorg.conf ]; then
   sed -i "s%.*Option.*XkbLayout.*%	Option      \"XkbLayout\" \"${XKB_LAYOUT}\"%" /etc/X11/xorg.conf
- fi
- if [ -f ~/.Xwaylandrc ]; then
-   sed -i "s%^XKB_DEFAULT_LAYOUT=.*%XKB_DEFAULT_LAYOUT=$XKB_LAYOUT%" ~/.Xwaylandrc
- fi
- if [ -f "$XDG_CONFIG_HOME/labwc/environment" ]; then
+  FLAG_CHANGED="${FLAG_CHANGED}$(gettext 'Keyboard Layout'):yes|" #111020 no need to restart X.
+ elif [ -f ~/.Xwaylandrc ]; then
```

Comment:
This one shouldn't be `if`, not `elif`. For each configuration file, quicksetup should check if it exists, and modify it. This will keep the settings synced, if the user moved from X.Org to Xwayland, or from X.Org to labwc. And some users keep old save files, so xorg.conf can be there while it's a labwc-based Puppy.